### PR TITLE
.github/workflows: Create docker release and publish it to ghcr

### PR DESCRIPTION
**Pull request by kfirtoledo**
Create Docker images and publish them to ghcr, every new release(tag).

### CODE REVIEW

File: .github/workflows/make-release.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    - name: Run docker build
+      run: make docker-build
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Build and push cl-dataplane image
+      run: |
```
Comment (elevran) [inline .github/workflows/make-release.yml:31]: does it make sense to have a make target for that? Similar to L23 above which runs `make docker-build`.
nit: also, the name on the above line (L30) should probably not include `build` in it as it only tags and pushes.

**Review by elevran — APPROVED** [review]

File: .github/workflows/make-release.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    - name: Build and push cl-cgo-dataplane image
+      run: |
+        docker tag cl-cgo-dataplane:latest ghcr.io/clusterlink-net/cl-go-dataplane:${{ github.ref_name }}
+        docker push ghcr.io/clusterlink-net/cl-go-dataplane:${{ github.ref_name }}
+
+    - name: Build and push gwctl image
+      run: |
+        docker tag gwctl:latest ghcr.io/kfirtoledo/gwctl:${{ github.ref_name }}
+        docker push ghcr.io/kfirtoledo/gwctl:${{ github.ref_name }}
```
Comment (zivnevo) [inline .github/workflows/make-release.yml:48]:
```suggestion
        docker tag gwctl:latest ghcr.io/clusterlink-net/gwctl:${{ github.ref_name }}
        docker push ghcr.io/clusterlink-net/gwctl:${{ github.ref_name }}
```

File: .github/workflows/make-release.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    name: Create docker release and publish to ghcr
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ['1.20']
+    steps:
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: ${{ matrix.go }}
```
Comment (zivnevo) [inline .github/workflows/make-release.yml:19]:
```suggestion
    steps:
    - name: Set up Go
      uses: actions/setup-go@v4
      with:
        go-version-file: ./go.mod
```
When releasing, better use the Go version specified in `go.mod`

File: .github/workflows/make-release.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+name:  Image release
+
+on:
+  push:
+    tags:
+      - '*'
```
Comment (zivnevo) [inline .github/workflows/make-release.yml:6]:
```suggestion
      - 'v**'
```
Maybe only release when the tag starts with `v` (to allow other tags as well)?

File: Makefile [hunk 5]
```diff
@@ -87,10 +86,15 @@ docker-build: build
...
-	docker push ${IMG}
+push-image: docker-build
+	docker tag cl-dataplane:latest ghcr.io/clusterlink-net/cl-dataplane:$(IMAGE_VERSION)
+	docker push ghcr.io/clusterlink-net/cl-dataplane:$(IMAGE_VERSION)
+	docker tag cl-controlplane:latest ghcr.io/clusterlink-net/cl-controlplane:$(IMAGE_VERSION)
+	docker push ghcr.io/clusterlink-net/cl-controlplane:$(IMAGE_VERSION)
+	docker tag cl-go-dataplane:latest ghcr.io/clusterlink-net/cl-go-dataplane:$(IMAGE_VERSION)
+	docker push ghcr.io/clusterlink-net/cl-go-dataplane:$(IMAGE_VERSION)
+	docker tag cl-gwctl:latest ghcr.io/clusterlink-net/cl-gwctl:$(IMAGE_VERSION)
+	docker push ghcr.io/clusterlink-net/cl-gwctl:$(IMAGE_VERSION)
```
Comment (zivnevo) [inline Makefile:97]: Use `$(IMAGE_BASE)` here too

**Review by zivnevo — APPROVED** [review]
LGTM

**Review by elevran — APPROVED** [review]

**Pull request merged**